### DISCUSSION TITLE
URI naming conventions

### DISCUSSION THREAD

**Discussion by lecoqlibre** [datafoodconsortium/taxonomies#13] [Ideas]
Would it be possible to replace the SKOS Concepts id containing a number like:
`"@id" : "http://static.datafoodconsortium.org/data/ProductGlossary_Measure.rdf#4Pack"`

By:
`"@id" : "http://static.datafoodconsortium.org/data/ProductGlossary_Measure.rdf#FourPack"`

(#4PACK to #FourPack)

Reason: the connector use these ids as variable names and they can't start by a number.

We can add a function to change these ids inside the connector itself but it could be more convenient to change these for everyone.

If you agree, I can change the ids with VocBench.

@simonLouvet @amardeilh @Alcoz @RaggedStaff @RachL

**Comment by RaggedStaff** [comment]
I'm fine with this. 

Can we have a general rule that we don't allow Concept Id's that start with a number (to facilitate the Connector's functionality)?

**Comment by RaggedStaff** [comment]
@lecoqlibre - Is this not done yet? 🤨 I can't see any issues with this. Pretty sure Python has a similar rule... guess it would be good to be really cleae about what we are okay with in Concept Id Anchors... something like:

1. Only contain alphanumerics & underscore (_)
2. Cannot begin with a numeric
3. Are case sensitive_ (are they? 🤔)_

Any more to add?

**Reply by RaggedStaff** [reply]
We don't want duplicates based on changes to case, so the Anchors should be **Case Insensitive** and generally quote in lowercase, with hyphens

**Comment by RaggedStaff** [comment]
Rules we've agreed (at 21/9/23 meeting) are:

1. Only contain alphanumerics & hyphens ( - )
2. MUST begin with an alpha character (not numeric or hyphen)
3. Are case INSENSITIVE, but should usually be given in lower case

**Reply by lecoqlibre** [reply]
As said in the issue https://github.com/datafoodconsortium/connector-typescript/issues/5 it might be a better idea to change the rules to:
1. Only contain alphanumerics.
2. MUST begin with an alpha character (not numeric).
3. Use PascalCase for classes and camelCase for properties.

For the list of a possible cases, see https://stackoverflow.com/questions/17326185/what-are-the-different-kinds-of-cases.

**Reply by RaggedStaff** [reply]
I agree. I don't think we got anywhere with implementing these but as @mkllnk says:we should do this asap to minimise impacts.

@Alcoz - What do you think ?

**Reply by RaggedStaff** [reply]
As @lecoqlibre pointed out in [this Issue](https://github.com/datafoodconsortium/connector-typescript/issues/5#issuecomment-1943613324) we can mitigate some of the impact of these changes with the use of the __depreciated__ tag, but this could still cause problems if some platforms are using the old tags & some the new. So I think it needs to be a major release. :confounded: 

I think we need to get this done asap, and try to ensure platforms move over as quickly as possible/be aware of the potential issue during implementations. 

@simonLouvet - how much impact will this have on the Prototype? Presumably you're just receiving values from the platforms ?  :thinking:

**Reply by simonLouvet** [reply]
no real impact for prototype. Taxonomy have to be reload in triplestore and data provided by example or platform have to reference good uri.

**Comment by lecoqlibre** [comment]
Today during the ontology call with @Alcoz and @RaggedStaff we said that taxons (from the taxonomies) should use PascalCase and follow the [general naming rules](https://github.com/datafoodconsortium/taxonomies/discussions/13#discussioncomment-8464997):
1. Only contain alphanumerics.
2. MUST begin with an alpha character (not numeric).
3. Use PascalCase for classes and taxons and camelCase for properties.
